darts/models/forecasting/regression_model.py

```diff
@@ -428,105 +352,61 @@ def fit(
-                self.lags_historical_covariates is None
-                and self.lags_future_covariates is None
-            ),
+            future_covariates is not None and "future" not in self.lags,
             "`future_covariates` not None in `fit()` method call, but `lags_future_covariates` is None in "
             "constructor.",
         )
 
         raise_if(
-            future_covariates is None
-            and (
-                self.lags_future_covariates is not None
-                or self.lags_historical_covariates is not None
-            ),
-            "`future_covariates` is None in `fit()` method call, but `lags_future_covariates` is not None in"
+            future_covariates is None and "future" in self.lags,
+            "`future_covariates` is None in `fit()` method call, but `lags_future_covariates` is not None in "
             "constructor.",
         )
 
-        # saving the input dim, so that we can perform the dim check at prediction time
-        series_dim = series.width if isinstance(series, TimeSeries) else series[0].width
-        covariates_dim = 0
-        for covariates in [past_covariates, future_covariates]:
-            if covariates is not None:
-                if isinstance(covariates, TimeSeries):
-                    covariates_dim += covariates.width
-                else:
-                    covariates_dim += covariates[0].width
-
-        self.input_dim = series_dim + covariates_dim
-
-        # if series is multivariate wrap model with MultiOutputRegressor
-        if not series[0].is_univariate:
-            self.model = MultiOutputRegressor(self.model, n_jobs=1)
+        # saving the dims of all input series to check at prediction time
+        if isinstance(series, TimeSeries):
+            self.input_dim = {
+                "target": series.width,
+                "past": past_covariates.width if past_covariates else None,
+                "future": future_covariates.width if future_covariates else None,
+            }
+        else:
+            self.input_dim = {
+                "target": series[0].width,
+                "past": past_covariates[0].width if past_covariates else None,
+                "future": future_covariates[0].width if future_covariates else None,
+            }
+
+        # if multi-target regression
+        if not series[0].is_univariate or self.output_chunk_length > 1:
+            # and model isn't wrapped already
+            if not isinstance(self.model, MultiOutputRegressor):
+                # check whether model supports multi-target regression natively
+                if not (
+                    callable(getattr(self.model, "_get_tags", None))
+                    and isinstance(self.model._get_tags(), dict)
+                    and self.model._get_tags().get("multioutput")
+                ):
+                    # if not, wrap model with MultiOutputRegressor
+                    self.model = MultiOutputRegressor(self.model, n_jobs=1)
```

Comment:
I wonder there could be some cases where `n_jobs > 1` might provide significant gains...

Reply:
right, I will test it out 👍